firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCapture.java

```diff
@@ -0,0 +1,432 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.common;
+
+import android.app.ActivityManager;
+import android.app.ActivityManager.RunningAppProcessInfo;
+import android.content.Context;
+import android.os.Build;
+import android.os.Build.VERSION;
+import android.os.Environment;
+import android.os.StatFs;
+import com.google.firebase.crashlytics.BuildConfig;
+import com.google.firebase.crashlytics.internal.model.Architecture;
+import com.google.firebase.crashlytics.internal.stacktrace.StackTraceTrimmingStrategy;
+import com.google.firebase.crashlytics.internal.stacktrace.TrimmedThrowableData;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class is responsible for capturing information from the system and exception objects,
+ * parsing them, and passing them to handlers implemented by callers to use the captured information
+ * as they see fit.
+ */
+public class CrashlyticsReportDataCapture {
+
+  /**
+   * Defines methods for handling static report data gathered from the device, as well as system and
+   * build properties.
+   */
+  public interface ReportDataHandler {
```

Comment:
I find this model extremely odd. Normally when you have a data structure, you have something like a struct with nested fields, and you have one method that creates or populates the object, and then you pass that object to methods that do something with it. But here, instead of creating the object, we call a handler for every field in the object that would be set. Why does this need to be so complicated? Can't we just populate a Report object and hand it back, and let the other code manipulate it like any other object?

Reply:
Discussed this offline and decided to take a different approach.